### DISCUSSION TITLE
Unique User Count With HT Persona Role

### DISCUSSION THREAD

**Discussion by Ashwiniev95** [Sunbird-Lern/Community#68] [Query] [answered]
Hi @reshmi-nair @abhign @tanweer555 

We have a requirement from Product team to get this metric as a Big number in Admin Dashboard, at state and district level granularity. attaching the template
[SL internal_ User's Data.pptx](https://github.com/Sunbird-Lern/Community/files/10479673/SL.internal_.User.s.Data.pptx)

> Metric : Number of unique users with HT persona role on platform

Few explorations :
1. Checked with a rollup Datasource **summary-distinct-counts** that provides total_unique_users metric. But here we have an issue, Since we need the unique count till date(consolidated count), we couldn't query this datasource which has huge data since it can  impact the performance.
2. Checked with user search API's **api/user/v1/search** . This gives count of users but can we use this API to get unique user count at district level granularity

Can you please  let us know what could be the better option to get this metric or should we create a new custom data-product that serves the purpose.

If we have any similar report existing, Please let us know, so that will check and use that report if it satisfies our requirement.

-CC @Shakthieshwari @vijiurs

**Comment by Shakthieshwari** [comment]
@reshmi-nair @rhwarrier @SanthoshVasabhaktula @kumarks1122 Can somebody help us on this please ?

Thanks

**Reply by reshmi-nair** [reply]
@Shakthieshwari Pls create a new data product for generating this report, user/search API and druid data-source will not be apt for this.. For reference you can check this link. 

[StateAdminGeoReport](https://github.com/Sunbird-Lern/data-products/blob/master/lern-data-products/src/main/scala/org/sunbird/userorg/job/report/StateAdminGeoReportJob.scala)

[UserOrg Reports](https://lern.sunbird.org/learn/product-and-developer-guide/user-and-org-service/reports)

**Comment by Ashwiniev95** [comment]
@reshmi-nair Thanks for confirming.

**Comment by Ashwiniev95** [comment]
Hi @reshmi-nair 

We need to get the unique count of user with HT & other official as a filter on there role. So can you please share any data product that queries the redis with metrices. So that it would be helpful

or Can you share us few sample data from redis and insertion method, based on that we will try our requirement on count

or Can we extract all user Id and perform calculation in data product itself?

Can you please help us out here

-CC @Shakthieshwari @vijiurs

Thanks 
Ashwini

**Reply by reshmi-nair** [reply, marked as answer]
@Ashwiniev95 We dont have a data product currently which creates a metric using redis data.

Please check this link for redis insertion details
[caching-and-denormalising-user-metadata](https://lern.sunbird.org/learn/product-and-developer-guide/user-and-org-service/caching-and-denormalising-user-metadata)

You can create a dataproduct which will read the data from redis and create metrics
